### PR TITLE
[WIP] Update C API to mxnet v1.4.0

### PR DESCRIPTION
**Pull request by daniel-zullo**

### CODE REVIEW

File: src/mxnet/c/c_api.d [hunk 1]
```diff
@@ -51,36 +53,51 @@ extern (C):
 /*! \brief MXNET_DLL prefix for windows */
 
 /*! \brief manually define unsigned int */
-alias uint mx_uint;
+alias mx_uint = uint;
```
Comment (alaksiej-stankievicx-dunnhumby) [inline src/mxnet/c/c_api.d:56]: I think change `alias` for more D2 syntax should be done in separate commit.

File: src/mxnet/c/c_api.d [hunk 2]
```diff
@@ -104,7 +121,13 @@ struct NDArrayOpInfo
     bool function (int, int*, uint**, void*) infer_shape;
     bool function (char***, void*) list_outputs;
     bool function (char***, void*) list_arguments;
-    bool function (const(int)*, const(int)*, const(int)*, int*, int**, void*) declare_backward_dependency;
+    bool function (
```
Comment (alaksiej-stankievicx-dunnhumby) [inline src/mxnet/c/c_api.d:124]: and this is also stylistic changes should be moved to separate commit.

**Review by alaksiej-stankievicx-dunnhumby — COMMENTED** [review]
You should split style update vs real API update, because now it is difficult to track it.

File: src/mxnet/c/c_api.d (hunk 1 above)
Comment (daniel-zullo) [reply src/mxnet/c/c_api.d:56]: yes, that's right. Jens explain me the way to update it and we will address previous versions first

File: src/mxnet/c/c_api.d (hunk 2 above)
Comment (daniel-zullo) [reply src/mxnet/c/c_api.d:124]: I have found the right `dstep` command arguments to avoid these changes

**Comment by daniel-zullo** [comment]
Closing as previous versions will be addressed first.
